boards/posix/native_posix/sdl_model.c

```diff
@@ -0,0 +1,61 @@
+/* Copyright (c) 2018 dXplore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "sdl_model.h"
+
+#include <SDL.h>
+#include <signal.h>
+
+#include "posix_trace.h"
+
+static void hwm_sdl_handle_window_event(const SDL_Event *event)
+{
+	SDL_Window *window;
+	SDL_Renderer *renderer;
+
+	switch (event->window.event) {
+	case SDL_WINDOWEVENT_EXPOSED:
+
+		window = SDL_GetWindowFromID(event->window.windowID);
+		if (window == NULL) {
+			return;
+		}
+
+		renderer = SDL_GetRenderer(window);
+		if (renderer == NULL) {
+			return;
+		}
+		SDL_RenderPresent(renderer);
+		break;
+	default:
+		break;
+	}
+}
+
+void hwm_sdl_init(void)
+{
+	if (SDL_Init(SDL_INIT_VIDEO) < 0) {
+		posix_print_error_and_exit("Failed to initialize SDL: %s\n",
+				SDL_GetError());
+	}
+}
+
+void hwm_sdl_poll_events(void)
+{
+	SDL_Event event;
+
+	while (SDL_PollEvent(&event)) {
+		switch (event.type) {
+		case SDL_WINDOWEVENT:
+			hwm_sdl_handle_window_event(&event);
+			break;
+		case SDL_QUIT:
+			raise(SIGTERM);
```

Comment:
why do you raise SIGTERM instead of calling `posix_exit`?

Reply:
Calling posix_exit would immediately exit the application and bypass the exit code in hwm_sleep_until_next_timer (currently only printing the simulated time)

Reply:
* `posix_exit()` is the exit point for native_posix, which carries all necessary pre-exit steps. The "stopped at" print in `hwm_sleep_until_next_timer()` is just a silly nicety (linked with the --stop-at command line option). (thinking loud: Maybe I should just remove that print if it confuses the intent..)
* The signals were hooked to allow people to Ctrl+C/kill and still close in a controlled manner. They are not meant to be a normal way of closing. (Though they could be used from, say, a possible future asynchronous thread if there would be no better option).
* Having the signal flag exit thru the same place as the stop_at was just for pseudo code economy reason.

Reply:
ok I will call posix_exit